### ISSUE TITLE
Getter/setter doesn't check method arity

### ISSUE THREAD

**Issue by I-asked** [digitalgust/miniJVM#29]
Hi everyone and Hi Gust, first of all, I wanted to thank you and everyone involved a lot for this awesome project!

To the point:

Say you have a method like this:

```java
int discardAndGetX(int x) {
    return this.x;
}
```

The getter optimization doesn't check the arity of the method and pops the integer instead of `this` from the stack, resulting in say an NPE.

I am not sure how to fix this without compromising the performance, but it's a real problem with legacy libraries that expose such API's for compatibility…

**Comment by digitalgust** [comment]
Yes ,thank you , this issue fixed. the issue that missed the parameter of getter method. setter has the same issue.
